src/rt-app.c

```diff
@@ -913,174 +913,185 @@ static int __sched_priority(sched_data_t *sched_data)
 	 return 0;
 }
 
-static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
+static void _set_thread_policy_priority(thread_data_t *data,
+					sched_data_t *sched_data)
 {
 	struct sched_param param;
-	policy_t policy;
+	int ret;
+
+	/* Priority unchanged => Policy unchanged */
+	if (sched_data->prio == THREAD_PRIORITY_UNCHANGED)
+		return;
+
```

Comment:
this is not enough, we can set new priority without changing policy. In the cfs case, we don't want to call pthread_setschedparam